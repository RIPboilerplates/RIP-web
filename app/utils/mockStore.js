import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { fromJS } from 'immutable'

// Add constants needed in default state for testing
import { INITIAL_STATE as nav } from 'redux/Navigation/constants'

const defaultState = {
  nav,
}
export { defaultState }

export default (initialState) => {
  const mockStore = configureStore([thunk])

  return mockStore(fromJS(initialState || defaultState))
}
