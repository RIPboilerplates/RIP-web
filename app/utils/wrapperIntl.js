import React from 'react'
import PropTypes from 'prop-types'
import { IntlProvider } from 'react-intl'

export default class Wrapper extends React.PureComponent {
  static defaultProps = {
    locale: 'en',
  }

  static propTypes = {
    locale:   PropTypes.string,
    children: PropTypes.any.isRequired,
  }

  render() {
    return (
      <IntlProvider locale={this.props.locale}>
        {this.props.children}
      </IntlProvider>
    )
  }
}
