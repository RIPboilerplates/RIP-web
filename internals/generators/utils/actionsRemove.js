const fs = require('fs')
const path = require('path')
const { exec } = require('child_process')
const print = require('../../scripts/helpers/print')

const delLine = (answers, config) => {
  const filePath = path.join(__dirname, '../../../', config.file)

  if (!config.file) throw Error('`file` is required')
  if (!config.key) throw Error('`key` is required')
  if (!fs.existsSync(filePath)) throw Error('file does not exist')

  const fileText = fs.readFileSync(filePath).toString().split('\n')
  const editedText = fileText.filter((line) => line.indexOf(config.key) < 0)

  fs.writeFileSync(filePath, editedText.join('\n'))
  return filePath.toString()
}

const delDir = (answers, config) => {
  const filePath = path.join(__dirname, '../../../', config.directory)

  if (!config.directory) throw Error('`directory` is required')
  if (!fs.existsSync(filePath)) throw Error('directory does not exist')

  const cmd = `if [ -d "${filePath}" ]; then rm -fr ${filePath}; fi`
  exec(cmd, (err, result, stderr) => {
    if (err || stderr) {
      throw err || stderr
    }
    print(result)
  })

  return filePath
}

module.exports = {
  delLine,
  delDir,
}
