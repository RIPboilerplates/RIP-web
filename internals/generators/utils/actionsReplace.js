const fs = require('fs')
const path = require('path')
const replace = require('replace-in-file')
const changeCase = require('change-case')

const renameDirectory = (answers, config) => {
  const to = changeCase.pascal(config.to)

  if (!config.dir) throw Error('`dir` is required')
  if (!config.from) throw Error('`from` is required')
  if (!to) throw Error('`to` is required')

  const base = path.join(__dirname, '../../../app/', config.dir)
  const fromPath = path.join(base, config.from)
  const toPath = path.join(base, to)

  try {
    fs.renameSync(fromPath, toPath)
  } catch (e) {
    throw e
  }

  return `${fromPath} => ${toPath}`
}

const renameInFiles = (answers, config) => {
  const to = changeCase.pascal(config.to)

  if (!config.dir) throw Error('`dir` is required')
  if (!config.from) throw Error('`from` is required')
  if (!to) throw Error('`to` is required')

  const base = path.join(__dirname, '../../../app/', config.dir)
  const fromPath = path.join(base, config.from)

  let changes = []
  try {
    changes = replace.sync({
      files: [
        `${fromPath}/*`,
        `${fromPath}/tests/*`,
        `${base}/index.js`,
      ],
      from: new RegExp(config.from, 'g'),
      to,
    })
  } catch (e) {
    throw e
  }

  const spacer = '\n          '
  return `${spacer}${changes.join(spacer)}`
}

module.exports = {
  renameInFiles,
  renameDirectory,
}
