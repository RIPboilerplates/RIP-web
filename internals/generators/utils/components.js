/**
 * componentExists
 *
 * Check whether the given component exist in either the components or containers directory
 */

const fs = require('fs')
const path = require('path')

const componentsList = fs.readdirSync(path.join(__dirname, '../../../app/components'))
const containersList = fs.readdirSync(path.join(__dirname, '../../../app/containers'))

componentsList.pop()
containersList.pop()

const components = componentsList.concat(containersList)

const componentExists = (comp) => components.indexOf(comp) >= 0

module.exports = {
  componentExists,
  componentsList,
  containersList,
}
