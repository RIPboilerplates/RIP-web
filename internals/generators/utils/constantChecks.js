const fs = require('fs')
const path = require('path')
const constantCase = require('constant-case')
const root = __dirname
const redux = '../../../app/redux'
const reduxers = fs.readdirSync(path.join(root, redux)).filter((f) => fs.statSync(path.join(root, redux, f)).isDirectory())

let privateReduxer

/**
 * reducerExists
 *
 * Check reduxer exists in redux directory
 * @param {string} reduxer - Reduxer to check if exists
 * @return {boolean} Is there a folder same as the @param
 */
const reducerExists = (reduxer) => {
  privateReduxer = reduxer
  return reduxers.indexOf(reduxer) < 0
}

/**
 * constantExists
 *
 * Check Constant exists in reduxer
 * @param {string} reduxer - Reduxer to check
 * @param {string} constant - If exists in Reducer
 * @return {boolean} Does the param match a current constant
 */
const constantExists = (constantToCheck) => {
  const file = path.join(root, redux, privateReduxer, '/constants.js')
  const data = fs.readFileSync(file)
  return data.indexOf(constantCase(constantToCheck)) >= 0
}

module.exports = {
  reducerExists,
  constantExists,
}
