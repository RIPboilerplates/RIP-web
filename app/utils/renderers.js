import React from 'react'
import renderer from 'react-test-renderer'
// I18n rendering
import { IntlProvider } from 'react-intl'
import { shallowWithIntl, mountWithIntl } from 'enzyme-react-intl'
// Redux rendering
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import mockStore from './mockStore'
import history from './history'

const createComponentWithIntl = (children, props = { locale: 'en' }) =>
  renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  )

export const renderWithRouter = (children, store = mockStore({})) =>
  renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {children}
      </ConnectedRouter>
    </Provider>
  )

export const shallow = (Component) => shallowWithIntl(Component).first().shallow()
export const mount = (Component) => mountWithIntl(Component).mount()

export default createComponentWithIntl
