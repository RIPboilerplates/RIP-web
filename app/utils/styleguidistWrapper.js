import React from 'react'
import PropTypes from 'prop-types'
import { IntlProvider } from 'react-intl'

import { Provider } from 'react-redux'
import { defaultState } from './mockStore'

import configureStore from '../store'

export default class Wrapper extends React.PureComponent {
  static defaultProps = {
    locale: 'en',
  }

  static propTypes = {
    locale:   PropTypes.string,
    children: PropTypes.any.isRequired,
  }

  render() {
    const store = configureStore(defaultState)
    return (
      <Provider store={store}>
        <IntlProvider locale={this.props.locale}>
          {this.props.children}
        </IntlProvider>
      </Provider>
    )
  }
}
