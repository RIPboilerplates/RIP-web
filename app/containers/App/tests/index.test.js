import React from 'react'
import { renderWithRouter } from 'utils/renderers'

import Component from '../'

jest.mock('navigation', () => 'RootNav')

describe('App Component', () => {
  describe('Renders', () => {
    it('default', () => {
      const component = renderWithRouter(<Component />).toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
