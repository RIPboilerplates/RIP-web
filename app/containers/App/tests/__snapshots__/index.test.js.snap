// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders default 1`] = `
<div
  className="styles__Container-eZxbLb fCoiuO"
>
  <RootNav />
</div>
`;
